src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -37,15 +36,16 @@ internal class ConfigurationClientManager : IConfigurationClientManager
         private readonly ConfigurationClientOptions _clientOptions;
         private readonly bool _replicaDiscoveryEnabled;
         private readonly SrvLookupClient _srvLookupClient;
+        private readonly string _validDomain;
 
         private IList<ConfigurationClientWrapper> _dynamicClients;
-        private IEnumerable<string> _validDomains;
         private DateTimeOffset _lastFallbackClientRefresh = default;
         private DateTimeOffset _lastFallbackClientRefreshAttempt = default;
         private Logger _logger = new Logger();
 
         private static readonly TimeSpan FallbackClientRefreshExpireInterval = TimeSpan.FromHours(1);
         private static readonly TimeSpan MinimalClientRefreshInterval = TimeSpan.FromSeconds(30);
+        private static readonly string[] TrustedDomainPrefixes = new[] { ".azconfig", ".appconfig" };

```

Comment:
It's not a prefix. `azconfig` and `appconfig` are both "label"s in the domain name.